plugins/module_utils/foundation/progress.py

```diff
@@ -24,7 +24,7 @@ def get(self, uuid):
     def wait_for_completion(self, uuid):
         state = ""
         delay = 30
-        timeout = time.time() + self.module.params["timeout"]
+        end_time = time.time() + (max(3600, self.module.params["timeout"]))
```

Comment:
i prefer the variable name as timeout because when you name it as end_time, invariably user reading code will look for start_time.